sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -116,17 +116,30 @@ case class CreateViewCommand(
         CommandUtils.uncacheTableOrView(sparkSession, name.quotedString)
       }
       val aliasedPlan = aliasPlan(sparkSession, analyzedPlan)
```

Comment:
we should call `checkCyclicViewReference` for temp view as well, and add test for it (or find the existing test for this and move it to the new suite).